src/SonarScanner.MSBuild.PreProcessor/TeamBuildPreProcessor.cs

```diff
@@ -122,11 +122,21 @@ private async Task<bool> DoExecute(ProcessedArgs localSettings)
             }
 
             var server = this.factory.CreateSonarQubeServer(localSettings);
-            if (!await server.IsServerLicenseValid())
+
+            try
             {
-                this.logger.LogError(Resources.ERR_UnlicensedServer, localSettings.SonarQubeUrl);
+                if (!await server.IsServerLicenseValid())
+                {
+                    this.logger.LogError(Resources.ERR_UnlicensedServer, localSettings.SonarQubeUrl);
+                    return false;
+                }
+            }
+            catch (Exception ex)
+            {
+                this.logger.LogError(ex.Message);
```

Comment:
I don't understand, why it should help? When you run it on CE with forced authentication then you got the exception and here you return false. So it's the "same" behaviour like previously. Does not?

Reply:
Not sure to understand. Here's a quick recap on how it should behave : 
- CE edition, forced auth : 401 if no credential provided
- CE edition, forced auth : 404 if credentials provided, in any case
- CE edition, no forced auth : 404 in any case

- Commercial editions, forced auth, if no credentials provided : 401
- Commercial editions, forced auth, if credentials provided : 200 if license if present (whether if it's valid or not), 404 if license has not been found
- Commercial editions, no forced auth, 200 if license if present (whether if it's valid or not), 404 if license has not been found

Reply:
Sorry, you changed the new httpclient to this.client and I missed it. So it's ok.